### PR TITLE
Output response when a web request fails

### LINKED ISSUE #4058: Output raw json on API request failure
Will make debugging a lot easier. Use case is for something like #4056.

### PR DESCRIPTION
**Pull request by peppy**
Closes #4058.

### CODE REVIEW

File: osu.Game/Online/API/APIRequest.cs [hunk 1]
```diff
@@ -102,6 +102,13 @@ public void Fail(Exception e)
             WebRequest?.Abort();
 
             Logger.Log($@"Failing request {this} ({e})", LoggingTarget.Network);
+
+            if (!(e is OperationCanceledException) && WebRequest?.ResponseStream?.Length < 1024)
+            {
+                // in the case we fail a request, spitting out the response in the log is quite helpful.
+                Logger.Log($"Failed response was \"{WebRequest.ResponseString}\"", LoggingTarget.Network);
```
Comment (smoogipoo) [inline osu.Game/Online/API/APIRequest.cs]: Is there a reason you didn't make this as part of `JsonWebRequest`?

Comment (peppy) [reply osu.Game/Online/API/APIRequest.cs]: Because it's potentially useful for all web requests.

Comment (smoogipoo) [reply osu.Game/Online/API/APIRequest.cs]: Then as part of `WebRequest`? My point is that this would be useful to have for projects other than osu! too.

Comment (peppy) [reply osu.Game/Online/API/APIRequest.cs]: Maybe?

**Pull request closed without merging**